New bitflag error

When using tsinfer  I'm now getting the following error:

```
../../../miniforge3/envs/tsdate/lib/python3.12/site-packages/tsinfer/inference.py:1965: in get_samples_tree_sequence
    new_flags = np.bitwise_and(tables.nodes.flags, ~tskit.NODE_IS_SAMPLE)
E   OverflowError: Python integer -2 out of bounds for uint32
```

I think we might need to coerce tskit.NODE_IS_SAMPLE to the correct dtype, i.e.

```python
new_flags = np.bitwise_and(tables.nodes.flags, ~tables.nodes.flags.dtype.type(tskit.NODE_IS_SAMPLE))
```

--- a/tsinfer/inference.py
+++ b/tsinfer/inference.py
@@ -2293,7 +2293,10 @@ class SampleMatcher(Matcher):
 
         # All true ancestors are samples in the ancestors tree sequence. We unset
         # the SAMPLE flag but keep other flags intact.
-        new_flags = np.bitwise_and(tables.nodes.flags, ~tskit.NODE_IS_SAMPLE)
+        new_flags = tables.nodes.flags
+        new_flags = np.bitwise_and(
+            new_flags, ~new_flags.dtype.type(tskit.NODE_IS_SAMPLE)
+        )
         tables.nodes.flags = new_flags.astype(np.uint32)
         sample_ids = list(self.sample_id_map.values())
         assert len(tables.nodes) == sample_ids[0]

